recipes-kernel/linux/linux-raspberrypi.inc

```diff
@@ -48,81 +52,6 @@ KERNEL_MODULE_AUTOLOAD += "${@bb.utils.contains("MACHINE_FEATURES", "pitft28r",
-    # Remove the config
-    CONF_SED_SCRIPT="$CONF_SED_SCRIPT /CONFIG_$1[ =]/d;"
-    if test "$2" = "n"
-    then
-        echo "# CONFIG_$1 is not set" >> ${B}/.config
-    else
-        echo "CONFIG_$1=$2" >> ${B}/.config
-    fi
-}
-
-config_setup() {
-    # From kernel.bbclass. Unfortunately, this is needed to support builds that
-    # use devtool. The reason is as follows:
-    #
-    # - In devtool builds, externalsrc.bbclass gets inherited and sets a list of
-    # SRCTREECOVEREDTASKS, which don't get run because they affect the source
-    # tree and, when using devtool, we want the developer's changes to be the
-    # single source of truth. kernel-yocto.bbclass adds do_kernel_configme to
-    # SRCTREECOVEREDTASKS, so it doesn't run in a devtool build., In a normal
-    # non-devtool build, do_kernel_configme creates ${B}.config.
-    #
-    # - Normally (e.g. in linux-yocto), it would be OK that do_kernel_configme
-    # doesn't run, because the first few lines of do_configure in kernel.bbclass
-    # populate ${B}.config from either ${S}.config (if it exists) for custom
-    # developer changes, or otherwise from ${WORDIR}/defconfig.
-    #
-    # - In linux-raspberrypi, we add do_configure_prepend, which tweaks
-    # ${B}.config. Since this runs *before* the kernel.bbclass do_configure,
-    # ${B}.config doesn't yet exist and we hit an error. Thus we need to move
-    # the logic from do_configure up to before our do_configure_prepend. Because
-    # we are copying only a portion of do_configure and not the whole thing,
-    # there is no clean way to do it using OE functionality, so we just
-    # copy-and-paste.
-    if [ "${S}" != "${B}" ] && [ -f "${S}/.config" ] && [ ! -f "${B}/.config" ]; then
-        mv "${S}/.config" "${B}/.config"
-    fi
-
-    # Copy defconfig to .config if .config does not exist. This allows
-    # recipes to manage the .config themselves in do_configure_prepend().
-    if [ -f "${WORKDIR}/defconfig" ] && [ ! -f "${B}/.config" ]; then
-        cp "${WORKDIR}/defconfig" "${B}/.config"
-    fi
-}
-
-do_configure_prepend() {
-    config_setup
-
-    mv -f ${B}/.config ${B}/.config.patched
-    CONF_SED_SCRIPT=""
-
-    if [ "${INITRAMFS_IMAGE_BUNDLE}" = "1" ]; then
-        kernel_configure_variable OVERLAY_FS y
-        kernel_configure_variable SQUASHFS y
-        kernel_configure_variable UBIFS_FS y
-    fi
-
-    # Activate the configuration options for VC4
-    VC4GRAPHICS="${@bb.utils.contains("MACHINE_FEATURES", "vc4graphics", "1", "0", d)}"
-    if [ "${VC4GRAPHICS}" = "1" ]; then
-        kernel_configure_variable I2C_BCM2835 y
-        kernel_configure_variable DRM y
-        kernel_configure_variable DRM_FBDEV_EMULATION y
-        kernel_configure_variable DRM_VC4 y
-    fi
-
-    # Keep this the last line
-    # Remove all modified configs and add the rest to .config
-    sed -e "${CONF_SED_SCRIPT}" < '${B}/.config.patched' >> '${B}/.config'
-    rm -f ${B}/.config.patched
-}
-
```

Comment:
Can you check if devtool workflow works with the kernel after these changes? its a good test

Reply:
Will do.

Reply:
The following works:

1) `devtool modify virtual/kernel`

2) Add `pr_notice("Modified with devtool\n");` after the initial banner is printed in start_kernel().

3) Commit the change.

4) `devtool update-recipe linux-raspberrypi` (modifies `linux-raspberrypi.inc` instead of `linux-raspberrypi_5.4.bb` but I think that's a devtool quirk rather than due to something wrong in meta-raspberrypi).

5) `devtool reset linux-raspberrypi && rm -rf workspace`

6) `bitbake core-image-minimal`, copy the image to an SD card.

7) Boot the board and see "Modified with devtool" in dmesg.

Is that enough of a test? I don't use devtool much myself tbh.

Reply:
thanks thats fine then